openedx/core/djangoapps/content/search/documents.py

```diff
@@ -369,6 +373,54 @@ def _collections_for_content_object(object_id: OpaqueKey) -> dict:
     return result
 
 
+def _units_for_content_object(object_id: OpaqueKey) -> dict:
+    """
+    Given an XBlock, course, library, etc., get the units for its index doc.
+
+    e.g. for something in Units "UNIT_A" and "UNIT_B", this would return:
+        {
+            "units":  {
+                "display_name": ["Unit A", "Unit B"],
+                "key": ["UNIT_A", "UNIT_B"],
+            }
```

Comment:
I would prefer a structure like
```
"units":  [
  { "display_name":  "Unit A", "key": "UNIT_A"},
  { "display_name":  "Unit B", "key": "UNIT_B"},
]
```
but it is ok to leave it as it is, to make it consistent with the "collections".

Reply:
I can't remember what it was, but there had to be a reason why we used that format for collections?

Reply:
I don't remember well. @navinkarkera, could you help us with that question since you implemented https://github.com/openedx/edx-platform/pull/35600?

Reply:
Sorry, didn't mean to turn this into a thread.. it's not something we need to worry about at this moment @navinkarkera @ChrisChV  😄